packages/block/test/block.spec.ts

```diff
@@ -61,6 +62,61 @@ tape('[Block]: block functions', function (t) {
     st.end()
   })
 
+  t.test('initialization -> hardforkByBlockNumber option', function (st) {
+    const customChains = [testnetMerge]
+    const common = new Common({ chain: 'testnetMerge', hardfork: Hardfork.Istanbul, customChains })
+
+    let block = Block.fromBlockData(
+      {
+        header: {
+          number: 12, // Berlin block
+        },
+      },
+      { common, hardforkByBlockNumber: true }
+    )
+    st.equal(block._common.hardfork(), 'berlin', 'should use hardforkByBlockNumber option')
+
+    block = Block.fromBlockData(
+      {
+        header: {
+          number: 20, // Future block
+        },
+      },
+      { common, hardforkByTD: 5001 }
```

Comment:
Might it also be worth adding a test case for `TD` equal to the threshold (so 5000 in this case)?

Reply:
Test cases can and should generally be substantially expanded on the continued road to POS, especially in the blockchain library.